add thermal dispatch func

rathod-b opened the pull request:
Add thermal dispatch function v1

rathod-b committed:
add thermal dispatch func

Add thermal dispatch function v1

NLaws commented on src/dispatch.jl:
```diff
@@ -150,3 +150,75 @@ function plot_electric_dispatch(d::Dict; title="Electric Systems Dispatch", save
+    end
+    
+    # x axis resolution is determined by length of T.
+    x_axis = DateTime(2021):Dates.Minute(Int(60*(8760/T))):DateTime(2021,12,31,23,45)
```
@rathod-b I like that the thermal dispatch plot has a date axis. Will you please add this to the electric dispatch? Also, we should not make the year hard-coded. Please make year an input to both dispatch functions, defaulting to 2021.

rathod-b replied:
Updated in [7e4301b](https://github.com/NREL/REoptPlots.jl/pull/1/commits/7e4301b825ce1f21328b023b3ba1262f3ac096b8)

NLaws commented on src/dispatch.jl:
```diff
@@ -150,3 +150,75 @@ function plot_electric_dispatch(d::Dict; title="Electric Systems Dispatch", save
+    if "ExistingChiller" in keys(d)
+        T = length(d["ExistingChiller"]["year_one_electric_consumption_series"])
+    else
+        T = 8760
```
perhaps we should make `T` a function argument too? There are so many different cases for which thermal technologies could be in the results dictionary that it would be onerous to handle all of them in if/else statements.

rathod-b replied:
Makes sense, will make it an input arg and default to 8760

rathod-b committed:
include T and year as input args

include T (timesteps per year) and years as input args and update plot electric dispatch input args. Update x axis in plot electric dispatch to use year instead of sequence of integers.

NLaws commented on src/dispatch.jl:
```diff
@@ -37,13 +37,13 @@ function plot_electric_dispatch(d::Dict; title="Electric Systems Dispatch", save
         yaxis_title_text = "kW"
     )
-    
-    eload = d["ElectricLoad"]["load_series_kw"]
```
sorry when I was referring to making `T` a function argument I was only talking about the thermal dispatch. For the electric dispatch we will always have a d["ElectricLoad"]["load_series_kw"] so there is no need to have a `T` argument. Please change the electric dispatch back to inferring the `T` from the d["ElectricLoad"]["load_series_kw"]

rathod-b committed:
Update dispatch.jl

Retain the ability to determine T based on len of electric load series for electric dispatch plot. remove T as an input argument

NLaws committed:
fix module export

NLaws committed:
fix dependencies

NLaws reviewed (approved):
I have not tested `plot_thermal_dispatch`, but `plot_electric_dispatch` works and I assume that you tested the thermal function.
